### ISSUE TITLE
SupCon without gradient for optimizing model

### ISSUE THREAD

**Issue by BossunWang** [hangyu94/Ada-CM#8]
If you using torch.no_grad(), the feature_u and feature_u2 have no gradient to update model's parameter
https://github.com/hangyu94/Ada-CM/blob/fc88567a5eb6278db08e28dad9f167e94cb9e390/main.py#L215
Then, the loss just a constant value.
https://github.com/hangyu94/Ada-CM/blob/fc88567a5eb6278db08e28dad9f167e94cb9e390/main.py#L234 

The correct way is from original code:
https://github.com/HobbitLong/SupContrast/blob/331aab5921c1def3395918c05b214320bef54815/main_supcon.py#L219

**Comment by hangyu94** [comment]
Thanks for your suggestion. In our experiments, the line 215 was commented out. Due to a timing issue, the code was not checked carefully when it was uploaded. I will further check this code and update the detail later. If you have other problems, welcome to contact me by email.
